swift/ql/lib/codeql/swift/regex/Regex.qll

```diff
@@ -28,6 +28,39 @@ private class ParsedStringRegex extends RegExp, StringLiteralExpr {
   RegexEval getEval() { result = eval }
 }
 
+/**
+ * A data-flow node where a regular expression object is created.
+ */
+abstract class RegexCreation extends DataFlow::Node {
+  /**
+   * Gets a dataflow node for the string that the regular expression object is
+   * created from.
+   */
+  abstract DataFlow::Node getStringInput();
+}
+
+/**
+ * A data-flow node where a `Regex` or `NSRegularExpression` object is created.
+ */
+private class StandardRegexCreation extends RegexCreation {
+  DataFlow::Node input;
+
+  StandardRegexCreation() {
+    exists(CallExpr call |
+      (
+        call.getStaticTarget().(Method).hasQualifiedName("Regex", ["init(_:)", "init(_:as:)"]) or
```

Comment:
Pre-existing: Looking at https://developer.apple.com/documentation/swift/regex#initializers, should we also match `init(verbatim:)`?

Reply:
Yeah, I had to think about this one a bit.  `init(verbatim:)` creates a regular expression that matches exactly the given string, e.g. `init(verbatim: ".*")` creates a regex object that matches the string `".*"` (not a regex that matches any string).

So it's not actually taking a regular expression as input, hence why it's not modelled here (I will clarify the QLDoc for the `RegexCreation` class on this point).  It _does_ create a `Regex` object, so there's an argument we should model it somehow ... though right now it would not be interesting for any existing or planned queries.

Reply:
Yeah, it could have a taint model.  I've made a note in one of my [many] lists of potential taint improvements!